Added Private Messaging Provider

slarkerino opened the pull request.

gdlcf88 commented on providers/PrivateMessaing/EasyAbp.NotificationService.Provider.PrivateMessaging/EasyAbp/NotificationService/Provider/PrivateMessaging/PrivateMessageNotificationSendingJob.cs:
```diff
@@ -0,0 +1,69 @@
+
+            var notificationInfo = await _notificationInfoRepository.GetAsync(notification.NotificationInfoId);
+
+            await _privateMessageAppService.CreateAsync(new EasyAbp.PrivateMessaging.PrivateMessages.Dtos.CreateUpdatePrivateMessageDto
```
Application service cannot be used here.

I have updated the PM module:
https://github.com/EasyAbp/PrivateMessaging/commit/e84cd76bf4f08cba0d804e03ce37daa184702d10

Now you can try to publish a [SendPrivateMessageEto](https://github.com/EasyAbp/PrivateMessaging/blob/e84cd76bf4f08cba0d804e03ce37daa184702d10/src/EasyAbp.PrivateMessaging.Domain.Shared/PrivateMessages/SendPrivateMessageEto.cs) to create a message.

gdlcf88 commented on providers/PrivateMessaing/EasyAbp.NotificationService.Provider.PrivateMessaging/EasyAbp/NotificationService/Provider/PrivateMessaging/PrivateMessageNotificationSendingJob.cs:
```diff
@@ -0,0 +1,69 @@
+                ToUserName = userName
+            });
+
+            await SaveNotificationResultAsync(notification, true);
```
Handle the [PrivateMessageSentEto](https://github.com/EasyAbp/PrivateMessaging/blob/e84cd76bf4f08cba0d804e03ce37daa184702d10/src/EasyAbp.PrivateMessaging.Domain.Shared/PrivateMessages/PrivateMessageSentEto.cs) event and set the notification result.

slarkerino replied:
We can not determine if the result of PM here by subscribing to PrivateMessageSentEto here. We will need something like a reference id when sending pm. By the info in PrivateMessageSentEto, we can only try to find notification by userid, but what if one user has send multiple messages at a time. @gdlcf88

gdlcf88 replied:
@slarkerino
OK, I will handle it :)

gdlcf88 replied:
@slarkerino

SendPrivateMessageEto is extensible now, see:
https://github.com/EasyAbp/PrivateMessaging/blob/master/src/EasyAbp.PrivateMessaging.Domain.Shared/PrivateMessages/SendPrivateMessageEto.cs

The `SendPrivateMessageEto.ExtraProperties` will be mapped to the `PrivateMessage.ExtraProperties`.

So after PM module v1.1.1, you can define a `NotificationId` property to the `PrivateMessage` entity to record the notification ID and get it from `PrivateMessageSentEto.ExtraProperties`.

gdlcf88 commented on providers/PrivateMessaing/EasyAbp.NotificationService.Provider.PrivateMessaging/EasyAbp/NotificationService/Provider/PrivateMessaging/CreatePrivateMessageNotificationEventHandler.cs:
```diff
@@ -0,0 +1,82 @@
+        {
+            foreach (var notification in notifications)
+            {
+                await _backgroundJobManager.EnqueueAsync(new PrivateMessageNotificationSendingJobArgs(notification.Id));
```
Why using a background job?

slarkerino commented:
OK Will do

slarkerino commented:
Currently we can't find a way to properly determine the referenced notification from the PrivateMessageSentEto. So I just by default assume all published pm results in success. To ensure a pm is sent, I think we have to rework the PrivateMessageModule to include something like a referenceid(in this case, notificationid), so we can use this id to refer to the actual notification and set the result.

slarkerino commented:
@gdlcf88 I Have reworked the private message module :P

gdlcf88 commented:
@slarkerino Please give me the Write permission of https://github.com/yuansicloud/NotificationService

slarkerino commented:
> @slarkerino Please give me the Write permission of https://github.com/yuansicloud/NotificationService

I have invited you :P

gdlcf88 commented:
@slarkerino Could you please make an `EasyAbp.NotificationService.Provider.PrivateMessaging.Tests` project?

slarkerino commented:
> @slarkerino Could you please make an `EasyAbp.NotificationService.Provider.PrivateMessaging.Tests` project?

OK, I have added and tested the unit test for private message module and fixed some naming issues for Sms Module

gdlcf88 commented on providers/PrivateMessaing/EasyAbp.NotificationService.Provider.PrivateMessaging.Test/EasyAbp/NotificationService/Provider/PrivateMessaging/NotificationServiceProviderMailingTestConsts.cs:
```diff
@@ -0,0 +1,11 @@
+
+namespace EasyAbp.NotificationService.Provider.PrivateMessaging
+{
+    public static class NotificationServiceProviderPrivateMessagingTestConsts
```
Unexpected file name.

gdlcf88 commented on providers/PrivateMessaing/EasyAbp.NotificationService.Provider.PrivateMessaging.Test/EasyAbp/NotificationService/Provider/PrivateMessaging/NotificationServiceProviderMailingTestModule.cs:
```diff
@@ -0,0 +1,31 @@
+        typeof(NotificationServiceProviderPrivateMessagingModule),
+        typeof(NotificationServiceDomainTestModule)
+    )]
+    public class NotificationServiceProviderPrivateMessagingTestModule : AbpModule
```
Unexpected file name.

slarkerino commented:
@gdlcf88  Fixed

gdlcf88 commented:
Thanks, @slarkerino.

slarkerino commented:
> Thanks, @slarkerino.

XD
